app/components/channel_drawer/channel_list.js

```diff
@@ -336,8 +463,27 @@ class ChannelList extends React.Component {
             );
         }
 
+        const modalOptions = (
+            <ModalOptions
+                title={this.state.modalTitle}
+                options={this.state.options}
+                visible={this.state.showOptions}
+                onChange={(option) => {
```

Comment:
The ModalOptions component re-renders once when the drawer is opened and then twice when a new channel is selected. Extending PureComponent did not change this behavior until this function was moved to a class method. Pretty small component to re-render, but just wanted to point it out if we're looking to make performance opts down the road.